editor/selectors.js

```diff
@@ -368,8 +377,47 @@ export function getEditedPostPreviewLink( state ) {
  * @param  {String} uid   Block unique ID
  * @return {Object}       Parsed block object
  */
-export function getBlock( state, uid ) {
-	return state.editor.blocksByUid[ uid ];
+export const getBlock = createSelector(
+	( state, uid ) => {
+		const block = state.editor.blocksByUid[ uid ];
+		const type = getBlockType( block.name );
+
+		if ( ! block || ! type || ! type.attributes ) {
+			return block;
+		}
+
+		const metaAttributes = reduce( type.attributes, ( result, value, key ) => {
+			if ( value && 'meta' in value ) {
+				result[ key ] = getPostMeta( state, value.meta );
+			}
+
+			return result;
+		}, {} );
+
+		// Avoid injecting an empty `attributes: {}`
```

Comment:
Is there a negative impact of injecting an empty attributes object? I think I might rather we normalize on the empty object.

Reply:
>The intent was to preserve the original behavior (returning an untouched item from `blocksByUid`) _unless_ there was any meta. Due to my lack of knowledge, this seemed preferable, in case any consumer of the selector was comparing object references.

Yeah, the intent became clearer in #2805 putting together the changes there, and I left the existing behavior. 👍